### ISSUE TITLE
BigBlueButton Mobile Apps

### ISSUE THREAD

**Issue by LukeStaszewski** [bigbluebutton/bigbluebutton#8732]
Hello,

Just wondering if there are any mobile apps in development at the moment?

If not I would be interested in developing some but if there is work being done on it already I would rather contribute towards that of course :)

**Comment by ffdixon** [comment]
There isn't anything in development at the moment.  We've focused on the HTML5 client for all platforms.

You could embed BigBlueButton within an android app and the embedded Chrome browser does support WebRTC.  However, on iOS, the embedded Safari Mobile browser does not.  You only get the WebRTC libraries on iOS when launching to Safari Mobile from your app.

**Comment by t-sharabi** [comment]
> Hello,
> 
> Just wondering if there are any mobile apps in development at the moment?
> 
> If not I would be interested in developing some but if there is work being done on it already I would rather contribute towards that of course :)

Hello
I would like to know if you did start the development or not and also if yes would you please send the link? 
thanks

**Comment by LaumiH** [comment]
Just a quick question, would this allow for screen sharing (especially on a tablet) aswell? I know that chrome on android does not support screen share.

**Comment by t-sharabi** [comment]
Dear All
My team started to develop a mobile application for the platform and would be happy to share with you all soon

**Comment by arsenieciprian** [comment]
Thank you

**Comment by NiceboyWiseboy** [comment]
Thank you

**Comment by LukeStaszewski** [comment]
Wow that's fantastic @t-sharabi !

**Comment by agussang** [comment]
> Dear All
> My team started to develop a mobile application for the platform and would be happy to share with you all soon

Can you create on your repository so my team can contribute too

**Comment by jihadaj** [comment]
@t-sharabi kindly share the repo of the mobile app

**Comment by Abhishek92kumar** [comment]
Has anyone achieved that Android App. I am really in need of this.

**Comment by Abhishek92kumar** [comment]
Please share even if its not fully functional

**Comment by a3c** [comment]
@t-sharabi can you share the repository of app

**Comment by pankajnegi1893** [comment]
hi we are using BigBlueButton in Web .  but when I am using the Meeting URL in Android it's working fine. But in iOS we are getting the issue. I am using React Native to achieve the result.

I am using below code :- 
`import {WebView} from 'react-native-webview';`

`<View style={{flex:1, width: '100%', height: '100%'}}> 
            <WebView
                key="comp1"
                ref={(ref) => {
                  this.webview = ref;
                }}
                useWebKit={true}
                userAgent={'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36'}
                source={{
                  uri: initialUrl,
                }}
                style={{flexGrow: 1}}
                allowStorage={false}
                cacheEnabled={false}
                originWhitelist={['*']}
                mixedContentMode={'compatibility'}
                javaScriptEnabled={true}
                cacheMode={'LOAD_NO_CACHE'}
                domStorageEnabled={true}
              /> 
              </View> `

For iOS, I have add the code below inside the method didFinishLaunchingWithOptions (before the line return YES;) in AppDelegate.m implementation:

`NSString *userAgent = @"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36";
NSDictionary *dictionary = [[NSDictionary alloc] initWithObjectsAndKeys:userAgent, @"UserAgent", nil];
[[NSUserDefaults standardUserDefaults] registerDefaults:dictionary];`


Above solution I havee applied but didn't get any success.  Please help me regarding above issue.

**Comment by matiasilva** [comment]
Follow the docs for development help: https://docs.bigbluebutton.org/2.2/getting-help.html

**Comment by bennyboer** [comment]
Hi there, we are currently attempting to implement a BBB app (https://github.com/bennyboer/bbb_app). Presentations, chats are working, but things like voice connections are still in progress.

**Comment by bennyboer** [comment]
Short update on the BBB app project (https://github.com/bennyboer/bbb_app): We have most things running including sharing the phones screen, receiving screen share streams, presentations, voice streams, voice input, chats, sharing the webcam, receiving webcam streams.
Currently we are working on a better phone UI.

Not implemented right now:
- Support for breakout rooms
- Shared notes
- Drawing using the phone on the whiteboard

**Comment by edmundlaugasson** [comment]
That's true - we need smart device (Android, iOS) app. Currently web browser version is not working properly - cannot join with audio - endless loop. Thank you for considering and moving toward real app!

**Comment by hostbbb** [comment]
@edmundlaugasson can you try connecting to https://test.bigbluebutton.org   make sure you are using chrome, firefox, or safari

Millions are using BBB daily,  browser not the issue...   If does not work, can post in bigbluebutton-setup on google, and the community will try to help you with server configuration.

**Comment by edmundlaugasson** [comment]
@hostbbb that's true - with test server it is working, using Brave browser in Android. I am not the manager of my own server. I already wrote to server manager. There in my server is Nextcloud + BBB used. This means, that BBB session permalink is created by appropriate [Nextcloud BBB app](https://apps.nextcloud.com/apps/bbb) - it might be that app issue. At least there is possible to get permalink BEFORE the session and we can share to participants the session link, that will not change and is known early before the session begins. As you know, we need to inform people in a timely manner BEFORE the session begins.

**Comment by xeruf** [comment]
The old app announcement, for reference: https://bigbluebutton.org/articles/bbb-mobile-now-available-on-android-marketplace/

The mentioned app seems to sadly be abandoned again though :/

**Comment by ffdixon** [comment]
Check out https://github.com/mconf/bbb-mobile-sdk from the good folks at Mconf.
